dask/bytes/core.py

```diff
@@ -498,34 +495,38 @@ def ensure_protocol(protocol):
             raise RuntimeError(msg)
         import hdfs3.dask  # register dask filesystem
 
-    elif protocol in _filesystems:
-        return
-
+    if protocol in _filesystems:
+        cls = _filesystems[protocol]
     else:
         raise ValueError("Unknown protocol %s" % protocol)
 
+    if storage_options is None:
+        storage_options = {}
+    fs = cls(**storage_options)
+    fs_token = tokenize(cls, protocol, storage_options)
+    return fs, fs_token
+
 
 _filesystems = dict()
-# see .local.LocalFileSystem for reference implementation
 
 
 class FileSystem(object):
-    def logical_size(self, path, compression):
-        if compression == 'infer':
-            compression = infer_compression(path)
-        if compression is None:
-            return self.size(path)
-        else:
-            with self.open(path, 'rb') as f:
-                f = SeekableFile(f)
-                g = seekable_files[compression](f)
-                g.seek(0, 2)
-                result = g.tell()
-                g.close()
-            return result
-
-    def get_block_locations(self, path):
-        return None, None, None
+    pass
```

Comment:
Do we test for subclassing this anywhere? Maybe remove it completely, or document that the intention is to subclass. I think removal is preferable, with the local file-system as the reference implementation.

Reply:
We don't check for subclassing anywhere, but `gcsfs` at least subclasses from it (others might as well). Will have to change that before removing fully. I'd also prefer full removal.